### DISCUSSION TITLE
Add option to make all bars the same fixed width

### DISCUSSION THREAD

**Discussion by fheyen** [CoderLine/alphaTab#1183] [Ideas]
Being able to simply set a fixed width for all bars would bring multiple advantages:
- You can enforce a layout that you know will work (but you might have to check/try for each file)
- The same bar should always look the same, as it's look will not depend on how far it is stretched due to neighboring bars
- On a page or endless layout, bars will align, making it easier to see repeating patterns
- (My use case:) Rendering multiple tab versions below each other makes it impossible to get bars aligned
- Maybe faster layouting (although performance is more than good enough for me)

Obviously, this would not look well with all tabs, but sometimes you know enough about it to choose a reasonable width, that might even work better than any automatic approach.

I imagine this to be easy to implement, so please consider it even if it does not seem to you that it brings much benefit.

**Comment by Danielku15** [comment]
This should be indeed fairly simple to implement in its basics and it has some reasonable use cases. 

I am just thinking if it would even make sense to not just set 1 width, but allow a more flexible adjustment of the music sheet. Applications like Guitar Pro and MuseScore allow a range of customizations where you can size each bar individually and even force line breaks at certain measures. 
![image](https://github.com/CoderLine/alphaTab/assets/674916/85b8470d-fb22-43c6-a726-cbc0677cf13d)

The sizes then stored in files are unfortunately often specific to the software to really work out nicely but having such a feature in alphaTab sounds nice and maybe there is at least some ratio that we could apply to translate the sizes of Software X to alphaTab. 

So for me it could be following changes: 

1. Extend the data model to hold hints like sizes (GP8 seem to have this on bar and even on beat level 😮) and force/prevent line breaks. 
2. Load the information from file formats like Guitar Pro. 
3. Add a setting which allows controlling whether alphaTab should try to respect those settings (false by default). 

Side Note: It should be possible to change those values quasi on-the-fly and request re-layout and painting to allow users in the UI to change values. 

For your concrete use cases you could then subscribe to the [`scoreLoaded`](https://alphatab.net/docs/reference/api/scoreloaded) event and simply fill those size hints with your fixed width for all bars according to your preference.

**Reply by goodgame365** [reply]
I have noticed that when there is only one audio track, this feature works fine. However, if there are two or more audio tracks, the layout display becomes disorganized.

**Comment by Danielku15** [comment]
@fheyen  The feature is available in 1.3.0-alpha.671

https://next.alphatab.net/docs/reference/settings/display/systemslayoutmode/

You would be interested into the horizontal layout + `displayWidth` combination. As described before you could hook into the `scoreLoaded` event and set the displayWidth for all your bars as preferered.
